openapi.yaml

```diff
@@ -164,353 +102,495 @@ paths:
+            - BC
+            - NT
+            - AB
+            - ON
+            - SK
+            - NB
+            - NF
+            - NS
+            - PE
+            - PQ
+            - MP
+            - FM
+            - MH
+            - PW
+            - MB
+            - YT
+            - AP
+            - AE
+            - AA
+            - FC
+            - FR
+            - NU
+            - BY
+            - ENG
+          description: Filter results by state code.
+        - name: filter[nation]
+          in: query
+          type: string
+          required: false
+          description: Filter results by nation.
+        - name: filter[nationCode]
+          in: query
+          type: string
+          required: false
+          description: Filter results by nation code.
         - $ref: '#/parameters/addressType'
-        - $ref: '#/parameters/osuID'
-        - $ref: '#/parameters/authorization'
       responses:
-        200:
+        '200':
           description: Successful response
           schema:
-            $ref: '#/definitions/AddressesResultObject'
-        404:
-          description: Person not found
-          schema:
-            $ref: '#/definitions/Error'
+            $ref: '#/definitions/AddressSetResult'
+        '400':
+          $ref: '#/responses/400'
+        '404':
+          $ref: '#/responses/404'
+        '500':
+          $ref: '#/responses/500'
     post:
       description: Create address for a person
       tags:
         - addresses
-      operationId: postAddresses
+      operationId: postAddress
       parameters:
         - name: addressBody
           in: body
-          description: Post body for the new address
+          description: Post body for new address
           required: true
           schema:
-            $ref: '#/definitions/NewAddressResultObject'
-        - $ref: '#/parameters/osuID'
-        - $ref: '#/parameters/authorization'
+            $ref: '#/definitions/AddressPostBody'
+        - $ref: '#/parameters/osuId'
       responses:
-        200:
+        '200':
           description: Successful response
           schema:
-            $ref: '#/definitions/AddressResultObject'
-        400:
-          description: Bad request
-          schema:
-            $ref: '#/definitions/ErrorArray'
-        404:
-          description: Person not found
-          schema:
-            $ref: '#/definitions/Error'
-  /persons/{osuID}/phones:
+            $ref: '#/definitions/AddressResult'
+        '400':
+          $ref: '#/responses/400'
+        '404':
+          $ref: '#/responses/404'
+        '409':
+          $ref: '#/responses/409'
+        '500':
+          $ref: '#/responses/500'
+  /persons/{osuId}/phones:
     get:
       description: Get phone numbers for a person
       tags:
         - phones
       operationId: getPhones
       parameters:
+        - $ref: '#/parameters/osuId'
         - $ref: '#/parameters/addressType'
         - $ref: '#/parameters/phoneType'
```

Comment:
What's the difference between phoneType and addressType here? Do we need both?

Reply:
@ianbrown9475 Yeah we need both of them. There are some overlaps between them, but there are still some are not. It's a very common use case that client needs to search by phoneType or addressType, so it'll be helpful if we have both of them implemented.